core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/PercentileRanksAggregationBuilder.java

```diff
@@ -98,6 +103,8 @@
     }
 
     public static AggregationBuilder parse(String aggregationName, XContentParser parser) throws IOException {
+        // the aggregation name is supplied to the parser as a Context, a bit hacky but it works well
```

Comment:
Thanks. I don't think its hacky ;-) Its okay as long as we have the context argument, when we try to remove it one day it might be useful to know why we use it here.